### ISSUE TITLE
Gnome-shell 3.26.2 vlc creates multiple entries

### ISSUE THREAD

**Issue by cesar-rickinho** [JasonLG1979/gnome-shell-extensions-mediaplayer#440]
When using VLC it creates multiple entries to the menu and eventually they fill up the screen.

OS: debian buster (testing)
VLC: 3.0.0-rc6
Gnome: 3.26.2
Media player indicator: 61

![screenshot from 2018-01-27 03-56-29](https://user-images.githubusercontent.com/11948362/35476078-c21713ca-03a1-11e8-97d6-e5185282dddf.png)

**Comment by JasonLG1979** [comment]
How many instances of VLC do you have running? Each instance has it's own MPRIS interface. The 1st instance will have a DBus name of org.mpris.MediaPlayer2.vlc and each one after that will have a DBus name of org.mpris.MediaPlayer2.vlc.instanceXXXX (where the XXXX is a number) Can you fire up d-feet and look for VLC and see if DBus shows more than one instance?

As a side note the latest version of the Extension is 63. Your version (61) is not supported see the Wiki:
 https://github.com/JasonLG1979/gnome-shell-extensions-mediaplayer/wiki/Bug-Reports-and-Feature-Requests#bug-reports

**Comment by cesar-rickinho** [comment]
Just updated the extension to version 63, one instance of VLC, multiple videos, for each video it creates one new entry on the menu.

Screenshot of d-feet
![image](https://user-images.githubusercontent.com/11948362/35476383-dc9a6d96-03a6-11e8-9d63-938a0fe37577.png)

**Comment by JasonLG1979** [comment]
> one instance of VLC, multiple videos

Multiple videos as in all at the same time or one after the other, just to be clear?

**Comment by JasonLG1979** [comment]
With identical version of GNOME Shell, VLC and the extension on Fedora 27 I can not repoduce your issue. What do the systemd logs say? You can open Logs go to the All side tab and search for "mediaplayer" and it should list any errors.

**Comment by JasonLG1979** [comment]
Could you also try disabling all other extensions. Sometimes extensions clash with each other.

**Comment by cesar-rickinho** [comment]
One instance of VLC playing, once a video stops and I start to play another (on the same instance) it creates a new option.

If I disable the media player extension all vlc entries disappear, if I enable the extension again it starts to do it again.

What logs should I be looging for on journalctl? (Don't wanna paste the logs here because it can be too much, pc has been running for 4 days :()

![screenshot from 2018-01-27 21-51-34](https://user-images.githubusercontent.com/11948362/35476708-5a662e0e-03ac-11e8-9015-fe1e1d50cda4.png)
![screenshot from 2018-01-27 21-52-04](https://user-images.githubusercontent.com/11948362/35476709-5a7bbe9a-03ac-11e8-9d0c-23ff0adf7fca.png)

**Comment by cesar-rickinho** [comment]
How likely do you think it's something on the Gnome 3 implementation (packages) of Debian?

**Comment by cesar-rickinho** [comment]
[file.txt](https://github.com/JasonLG1979/gnome-shell-extensions-mediaplayer/files/1670577/file.txt)

Logs file

**Comment by JasonLG1979** [comment]
Uncaught errors should get logged to the systemd log. The easy to check it is with the GUI app Logs.

**Comment by cesar-rickinho** [comment]
Here re the logs that I retrieved from gnome-logs.

[log.txt](https://github.com/JasonLG1979/gnome-shell-extensions-mediaplayer/files/1670598/log.txt)

**Comment by JasonLG1979** [comment]
> How likely do you think it's something on the Gnome 3 implementation (packages) of Debian?

If I had to guess it may be related to this: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=887082

**Comment by JasonLG1979** [comment]
The latest "stable" Gjs update introduced a pretty bad bug.

**Comment by JasonLG1979** [comment]
Other than that I don't see any errors from the extension from the logs. When I get some time maybe tomorrow(?) I'll fire up a Debian testing VM and see what I can come up with. Can you do me a favor and checkout this other MPRIS extension I wrote? It's much simpler so it''ll hopefully be easier to debug?
 https://github.com/JasonLG1979/gnome-shell-extensions-mpris-indicator-button

**Comment by cesar-rickinho** [comment]
Thanks for the feedback on the gnome-shell package, I'll wait for an update to see if it will be fixed.

This new extension works fine, the only thing i think that might be missing is the playback indicator (the one that show the time and progress of the media so that you can jump to a specific time or maybe skip something playing, not sure if that's the name). 

The error that I got from mediaplayer didn't happen on this new extension.

![screenshot from 2018-01-27 23-28-30](https://user-images.githubusercontent.com/11948362/35477419-e455e232-03b9-11e8-9064-25a8da3047d0.png)
Note: I had 2 vlc icons on the taskbar, but I can hide one of them.

**Comment by JasonLG1979** [comment]
The new extension is very simple on purpose. I don't really plan on adding any more features. It's an alternative to this extension not a replacement. I want to keep this open so it gets fixed. I'm curious as to why it effects Debian and not Fedora?

**Comment by cesar-rickinho** [comment]
Oh, ok then, no problem. 

Still prefer this one :), but thanks for the help and for your time 👍 .

**Comment by JasonLG1979** [comment]
The duplicate icon is I'm guessing another extension providing a system tray? The new extension will (try) to show the current player's icon, preferring a symbolic icon, but will use the full color icon if the theme does not have a symbolic icon for the player.

**Comment by cesar-rickinho** [comment]
Yes, the duplicate icon is because of another extension, so there is no need for you to worry about it.

**Comment by JasonLG1979** [comment]
>Still prefer this one 

Me too. But different strokes for different folks. I made the other one for more of a direct replacement for the Ubuntu soundmenu with 18.04 LTS coming out in a few months I figured a few people may miss it.

**Comment by cesar-rickinho** [comment]
>  I made the other one for more of a direct replacement for the Ubuntu soundmenu with 18.04 LTS coming out in a few months I figured a few people may miss it.
Got it, nice one

**Comment by cesar-rickinho** [comment]
>  I made the other one for more of a direct replacement for the Ubuntu soundmenu with 18.04 LTS coming out in a few months I figured a few people may miss it.

Got it

**Comment by JasonLG1979** [comment]
> Yes, the duplicate icon is because of another extension, so there is no need for you to worry about it.

https://github.com/Ubuntu/gnome-shell-extension-appindicator by chance? I noticed their icon was the wrong size and slightly blurry that's what gave it away. I tried to tell them how to make sure the icons are clear and the proper size no matter what theme by using the built in ```system-status-icon``` css style class instead of manually setting the size them self. Apparently no one listened...

**Comment by cesar-rickinho** [comment]
> https://github.com/Ubuntu/gnome-shell-extension-appindicator by chance?

No, it's the Topicons Plus (https://github.com/phocean/TopIcons-plus).

I don't like the Gnome default app indicator on the bottom corner, so I use that one to place them in the tray at the top (where they should be by default in my opinion).

**Comment by JasonLG1979** [comment]
I'm not really a huge fan of the appindicator API. It's a mess, way over complicated. My idea was to have a few purpose build DBus interfaces like MPRIS. You could have one for Chat apps that would let them expose common Chat app type info/functionality for example. It could then be left up to clients how to display/present that info/functionality to the user much in the way that MPRIS allows. MPRIS allows media players to expose info/functionality without having to care how it's presented or even if it is at all.

**Comment by fcastilloec** [comment]
I'm seeing this bug on Ubuntu 18.04 using `v63` of this extension. While checking the logs, as you mentioned previously, I only got this message 3 times:
> Ignored exception from dbus method: Gio.DBusError: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name org.mpris.MediaPlayer2.vlc.instance23421 was not provided by any .service files
And also got this message once:
> Error calling method GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name org.mpris.MediaPlayer2.vlc.instance23441 was not provided by any .service files

After closing all instances of VLC I still saw that something was being shown as playing from vlc even though there was no process for vlc. I tried disabling the extenstion to re-enable it, but my system froze and the gnome-shell just restarted itself.

If there's something I can do to help debug this, I'll be happy to do it. Just let me know.

**Comment by JasonLG1979** [comment]
@fcastilloec I am also running Ubuntu 18.04 at the moment and I can not reproduce your bug with VLC. Do other players work? Also is VLC a standard install or a flatpak/snap?

**Comment by martin31821** [comment]
I have the same problem on Arch with Gnome shell 3.30, VLC 3.0.4.
I have some log messages rgd Connecting signals on deallocated components, but that shouldn't matter.

A minor strange behavior, maybe important:
When stopping all VLC processes, then starting spotify, I get the correct album cover and running timeline in any of the displayed vlc indicators, but I can't control them.

**Comment by arthurflor** [comment]
> I have the same problem on Arch with Gnome shell 3.30, VLC 3.0.4.
> I have some log messages rgd Connecting signals on deallocated components, but that shouldn't matter.
> 
> A minor strange behavior, maybe important:
> When stopping all VLC processes, then starting spotify, I get the correct album cover and running timeline in any of the displayed vlc indicators, but I can't control them.

Also here. The extension create a VLC indicator to the each video in playlist. However, only the current video indicator is closed when I close the VLC.

**Comment by JasonLG1979** [comment]
I still can't reproduce. I'm on Fedora 28 (GNOME Shell 3.28) VLC 3.0.4. I have tried opening multiple videos in separate VLC instances. In that case it behaves as expected with multiple entries in the menu. I've also tried opening the same group of multiple videos in a single instance and I also get the expected behaviour, a single VLC entry.

Without some meaningful log info there's nothing I can really do if I can't reproduce it myself. I'm not saying that the info provided so far isn't helpful or that meaningful log info is even being generated by GNOME Shell. GNOME Shell logs can be very cryptic especially when it comes to extensions and are often time not very useful. But you understand of course.

@fcastilloec mentioned:
>org.mpris.MediaPlayer2.vlc.instance23441 was not provided by any .service files

That would suggest an issue with VLC's MPRIS implementation. If that's the case the best I can do is hopefully try to catch the errors and basically ignore the offending VLC instances.
